src/Service.Tests/OpenApiDocumentor/CLRtoJsonValueTypeUnitTests.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using Azure.DataApiBuilder.Core.Services;
+using Azure.DataApiBuilder.Core.Services.OpenAPI;
+using Azure.DataApiBuilder.Service.Exceptions;
+using Azure.DataApiBuilder.Service.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.OpenApiDocumentor
+{
+    /// <summary>
+    /// Validates TypeHelper converters return expected results.
+    /// </summary>
+    [TestClass]
+    public class CLRtoJsonValueTypeUnitTests
+    {
+        private const string ERROR_PREFIX = "The SqlDbType ";
+        private const string SQLDBTYPE_RESOLUTION_ERROR = "failed to resolve to SqlDbType.";
+        private const string JSONDATATYPE_RESOLUTION_ERROR = "(when supported) should map to a system type and associated JsonDataType.";
+        private const string DBTYPE_RESOLUTION_ERROR = "(when supported) should map to a system type and associated DbType.";
+
+        /// <summary>
+        /// Validates that all DAB supported CLR types (system types) map to a defined JSON value type.
+        /// A DAB supported CLR type is a CLR type mapped from a database value type.
+        /// </summary>
+        [TestMethod]
+        public void SupportedSystemTypesMapToJsonValueType()
```

Comment:
Rather than using a `foreach` loop in the test, you should use a data-drive test and provide the collection [as describe here](https://learn.microsoft.com/en-us/visualstudio/test/how-to-create-a-data-driven-unit-test?view=vs-2022#member-data-driven-test).

This makes it more obvious when a particular test fails, as they will be named more appropriately, and simplifies debugging, as you can be more targeted on debugging only the failing test.

Reply:
Thanks for the suggestion. I updated this to a DynamicData test and the tests are all distinct with readable test names. Makes it way easier to troubleshoot and identify which test case is running as you pointed out.